src/bin/fish.rs

```diff
@@ -274,11 +275,27 @@ fn source_config_in_directory(parser: &Parser, dir: &wstr) {
     parser.libdata_mut().pods.within_fish_init = true;
     let _ = parser.eval(&cmd, &IoChain::new());
     parser.libdata_mut().pods.within_fish_init = false;
+    return true;
 }
 
 /// Parse init files. exec_path is the path of fish executable as determined by argv[0].
 fn read_init(parser: &Parser, paths: &ConfigPaths) {
-    source_config_in_directory(parser, &str2wcstring(paths.data.as_os_str().as_bytes()));
+    let datapath = str2wcstring(paths.data.as_os_str().as_bytes());
+    if !source_config_in_directory(parser, &datapath) {
+        // If we cannot read share/config.fish, our internal configuration,
+        // something is wrong.
+        // That also means that our functions won't be found,
+        // and so any config we get would almost certainly be broken.
+        let escaped_pathname = escape(&datapath);
+        FLOGF!(
+            error,
+            "Fish cannot find its asset files in '%ls'.\n\
+             Please copy the share/ directory from the fish repository there and restart fish.\n\
```

Comment:
Wording nit: 

```suggestion
             Please copy the contents of the share/ directory from the fish repository there and restart fish.\n\
```

Reply:
Oh, we're gonna be arguing about the wording *a bunch*. This is very much preliminary, a rough draft.

Personally, I would make it:

```
Please download https://fishshell.com/data/fish-$version.tar.xz and extract it to $datadir
```

But that requires us to

1. decide that this is a way of installation worth supporting
2. commit to uploading tarballs that contain just share/ to fishshell.com

There are a bunch of other ways to go about this, including embedding all our scripts with something like [rust-embed](https://lib.rs/crates/rust-embed), potentially in a second binary (`fish_installer`), and then tell you to extract them.

I would like to come to an agreement on that bigger picture first.